app/models/post.rb

```diff
@@ -12,6 +12,15 @@ class Post < ApplicationRecord
 
   def publish!
     self.status = :published
+    self.published_at = Time.now
```

Comment:
Using `Time.now` can lead to problems with timezones, [it's a good practice to use `Time.current` instead](http://api.rubyonrails.org/classes/Time.html#method-c-current)